src/Features/LanguageServer/Protocol/Handler/Diagnostics/BuildOnlyDiagnosticIdsHandler.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler;
+
+[DataContract]
+internal readonly record struct BuildOnlyDiagnosticIdsResult([property: DataMember(Name = "ids")] string[] Ids);
+
+[ExportCSharpVisualBasicStatelessLspService(typeof(BuildOnlyDiagnosticIdsHandler)), Shared]
+[Method(BuildOnlyDiagnosticIdsMethodName)]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal class BuildOnlyDiagnosticIdsHandler(
```

Comment:
Do we have tests for the new handler?

Reply:
Thanks, added now.

I had looked at couple of other existing handlers and did not find any unit tests for them ([FoldingRangesHandler](https://sourceroslyn.io/#Microsoft.CodeAnalysis.LanguageServer.Protocol/Handler/FoldingRanges/FoldingRangesHandler.cs,56fd7d4d446838d4,references), [GoToDefinitionHandler](https://sourceroslyn.io/#Microsoft.CodeAnalysis.LanguageServer.Protocol/Handler/Definitions/GoToDefinitionHandler.cs,2ac6cb22e6f7aa29,references)). Do we have a tracking work item to audit all existing handlers and ensure we have unit tests for all of them? Thanks!